src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -70,25 +71,43 @@ function MetadataSidebarRedesign({
     const { formatMessage } = useIntl();
 
     const [selectedTemplates, setSelectedTemplates] = React.useState<Array<MetadataTemplate>>([]);
-    const [editingTemplate, setEditingTemplate] = React.useState<MetadataTemplateInstance | MetadataTemplate | null>(
-        null,
-    );
+    const [editingTemplate, setEditingTemplate] = React.useState<MetadataTemplateInstance | null>(null);
     const [isUnsavedChangesModalOpen, setIsUnsavedChangesModalOpen] = React.useState<boolean>(false);
 
-    const { file, templates, errorMessage, status, templateInstances } = useSidebarMetadataFetcher(
-        api,
-        fileId,
-        onError,
-        isFeatureEnabled,
-    );
+    const { createMetadataInstance, file, templates, errorMessage, status, templateInstances, updateMetadataInstance } =
+        useSidebarMetadataFetcher(api, fileId, onError, isFeatureEnabled);
 
     const handleUnsavedChanges = () => {
         setIsUnsavedChangesModalOpen(true);
     };
 
+    const convertToInstance = (template: MetadataTemplate): MetadataTemplateInstance => {
+        return {
+            canEdit: !!file.permissions.can_upload,
+            displayName: template.displayName,
+            hidden: template.hidden,
+            id: template.id,
```

Comment:
`template.id` and template instance id are not meant to be the same

Reply:
ok true, so what we want to set here when converting?

Reply:
we should generate one, similarly to how it currently happens for custom metadata
https://github.com/box/box-ui-elements/blob/fe4fededd993c4025b1b6f38cc8a6387af1a9acc/src/api/Metadata.js#L143